File: LeetSync/3799-unique-3-digit-even-numbers/unique-3-digit-even-numbers.py
class Solution:
    def totalNumbers(self, digits: list[int]) -> int:
        freq = Counter(digits)
        distinct_count = 0
        
        for num in range(100, 1000, 2):
            hundred = num // 100
            ten = (num // 10) % 10
            unit = num % 10
            
            num_freq = Counter([hundred, ten, unit])
            
            if all(num_freq[d] <= freq[d] for d in num_freq):
                distinct_count += 1
                
        return distinct_count

File: LeetSync/3799-unique-3-digit-even-numbers/unique-3-digit-even-numbers.cpp
class Solution {
public:
    int totalNumbers(vector<int>& digits) {
        vector<int>freq(10, 0);

        for(int digit: digits){
            freq[digit]++;
        }

        int count = 0;

        for(int num = 100; num < 999; num += 2){
            int hundred = num/100;
            int ten = (num / 10) % 10;
            int unit = num % 10;

            vector<int>currentFreq(10, 0);
            currentFreq[hundred]++;
            currentFreq[ten]++;
            currentFreq[unit]++;


            bool valid = true;
            for(int i = 0; i < 10; i++){
                if(currentFreq[i] > freq[i]){
                    valid = false;
                    break;
                }
            }

            if(valid){
                count++;
            }            
        }

        return count;
    }
};